rotkehlchen/db/dbhandler.py

```diff
@@ -2768,27 +2768,49 @@ def get_latest_asset_value_distribution(self) -> List[DBAssetBalance]:
             ignored_assets = self.get_ignored_assets(write_cursor)
 
         with self.conn.read_ctx() as cursor:
+            treat_eth2_as_eth = self.get_settings(cursor).treat_eth2_as_eth
             cursor.execute(
                 'SELECT time, currency, amount, usd_value, category FROM timed_balances WHERE '
                 'time=(SELECT MAX(time) from timed_balances) AND category = ? ORDER BY '
                 'CAST(usd_value AS REAL) DESC;',
                 (BalanceType.ASSET.serialize_for_db(),),  # pylint: disable=no-member
             )
             asset_balances = []
+            eth_balance: Dict[str, str] = {
+                'time': '0',
+                'amount': '0',
+                'usd_value': '0',
+            }
             for result in cursor:
                 asset = Asset(result[1])
                 if asset in ignored_assets:
                     continue
+                # show eth & eth2 as eth in value distribution by asset
+                if treat_eth2_as_eth is True and asset in (A_ETH, A_ETH2):
+                    eth_balance['time'] = str(result[0])
+                    eth_balance['amount'] = str(FVal(eth_balance['amount']) + FVal(result[2]))
+                    eth_balance['usd_value'] = str(FVal(eth_balance['usd_value']) + FVal(result[3]))  # noqa: E501
+                else:
+                    asset_balances.append(
+                        DBAssetBalance(
+                            time=result[0],
+                            asset=asset,
+                            amount=result[2],
+                            usd_value=result[3],
+                            category=BalanceType.deserialize_from_db(result[4]),
+                        ),
+                    )
+            # only add the eth_balance if it contains a balance > 0
+            if FVal(eth_balance['amount']) > ZERO or FVal(eth_balance['usd_value']) > ZERO:
                 asset_balances.append(
                     DBAssetBalance(
-                        time=result[0],
-                        asset=asset,
-                        amount=result[2],
-                        usd_value=result[3],
-                        category=BalanceType.deserialize_from_db(result[4]),
+                        time=Timestamp(int(eth_balance['time'])),
+                        asset=A_ETH,
+                        amount=eth_balance['amount'],
+                        usd_value=eth_balance['usd_value'],
+                        category=BalanceType.ASSET,
```

Comment:
Also you are duplicating code here. Maybe you can save all the info in some kind of list and the create the `DBAssetBalance` from that clean list avoiding duplication

Reply:
I don't totally get you here

Reply:
I was thinking in how to remove the duplication of the object creation but it would mean to go through the data twice... Does any other good idea come to your mind? otherwise I guess is fine

Reply:
Also, it's not duplication as I'm only adding the ETH balance after the loop. Unless I don't get you